Assets/Scripts/Robots/RouteMovement.cs

```diff
@@ -0,0 +1,135 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEditor;
+using System.Linq;
+using UnityEngine.Events;
+using UnityEditor.Callbacks;
+
+namespace Cosmobot
+{
+    public class RouteMovement : MonoBehaviour
+    {
+        public float robotSpeed;
+        public float gravity = 50;
+        public float distanceToGround = 0.1f;
+        public Vector3 direction = Vector3.zero;
+
+        #region route
+        public bool loopMode;
+        public int routeIndex = 0;
+        public Route[] route;
+        private bool routeForward = true;
+        #endregion route
+
+        private Rigidbody rb;
+        private Grabber grabber;
+
+        void Start()
+        {
+            rb = GetComponent<Rigidbody>();
+            grabber = GetComponentInChildren<Grabber>();
+        }
```

Comment:
add `[RequireComponent(typeof(Rigidbody), typeof(Grabber))]` attribute to class or add null check here